### ISSUE TITLE
The performance of Swin-Transformer in crowd scene

### ISSUE THREAD

**Issue by marvelcell** [hasanirtiza/PedesFormer#8]
Hello, I am very sorry to bother you, your work has helped me a lot. I am still researching the performance of Swin-transformer in detecting dense pedestrians and trying to improve its performance. I used CrowdHuman's entire training and validation set to train Swin-transformer+Cascade-Rcnn, but the mAP was 44.7, AP50 was 76.7, and AP75 was 45.1. The learning rate is 0.0001 at the beginning and will decay by 0.1 on the 27th and 35th epochs. May I ask if this is a good result in this model? Does it have any hope of further improvement? Thank you very much. #6

**Comment by hasanirtiza** [comment]
It is a good question, in my experience what I have seen is Swin requires a lot of training data (even more than crowdhuman ), may be at the scale of COCO. You can pre-train the network on WiderPedestrian (the data provided here (https://competitions.codalab.org/competitions/20132)). Then you can fine tune for any dataset that Swin model.

**Comment by marvelcell** [comment]
Thank you for your quick reply. I am very grateful for your suggestion, and your useful suggestions in issue6 have helped me graduate from the university in July!
But I am wondering, if the Swin-transformer requires a lot of training data, and needs to be pre-trained on WiderPedestrian, how can I compare it with other models on CrowdHuman? I mean, it seems that some models don't need to be pre-trained and can gain good results on CrowdHuman. 
Thank you.

**Comment by hasanirtiza** [comment]
> Thank you for your quick reply. I am very grateful for your suggestion, and your useful suggestions in issue6 have helped me graduate from the university in July! 

Congratulations on your graduation !

> But I am wondering, if the Swin-transformer requires a lot of training data, and needs to be pre-trained on WiderPedestrian, how can I compare it with other models on CrowdHuman? I mean, it seems that some models don't need to be pre-trained and can gain good results on CrowdHuman. Thank you.

Again it is an intricate question, I would refer you to read section 6 of this [paper](https://arxiv.org/pdf/2201.03176.pdf). We discuss some of the limitations of transformers with respect to CNNs and to summarize we summarize in the end by saying:

"We attribute this to the fact that potentially one of the main reasons are that the hyperparameters of transformer networks are not as optimized as they for CNNs. Moreover, transformer networks are also more prone to overfitting compared with the CNNs. Nonetheless, it is still relatively early for transformer networks compared with CNNs, which have been around and used extensively for nearly a decade and their hyperparameters are optimized quite thoroughly". 

Now this could one of the potential reason not the only reason. It would be an interesting research question to conduct experiments on this in a control environment. Hope that helps.

**Comment by marvelcell** [comment]
Thank you. I think it's an interesting question too, but it might be a little difficult for me to investigate it currently. And for the occluded pedestrian detection problem(especially on the CrowdHuman dataset), maybe it's better for me to use a better transformer-based model and train it with the CrowdHuman dataset. Thank you for answering my questions patiently.
